lab4/viewpoints/development/development.tex

```diff
@@ -0,0 +1,131 @@
+\section{Development View}
+
+\subsection{Module Organization}
+
+The EduPal quiz integration feature is organized into several interconnected modules, each responsible for specific aspects of the system:
+
+\subsubsection{User Interface Module}
+
+\begin{itemize}
+  \item \textbf{QuizPage}: The main page where users can start the quiz.
+  \item \textbf{QuestionPage}: The page that displays quiz questions to the users.
+  \item \textbf{ResultPage}: The page that shows the results after the quiz is completed.
+\end{itemize}
+
+\subsubsection{Quiz Session Module}
+
+\begin{itemize}
+  \item \textbf{QuizSessionController}: Manages user interactions during a quiz session.
+  \item \textbf{QuizSessionService}: Handles the business logic associated with a quiz session.
+  \item \textbf{QuizSessionRepository}: Manages the data storage and retrieval for quiz sessions.
+\end{itemize}
+
+\subsubsection{Quiz Management Module}
+
+\begin{itemize}
+  \item \textbf{QuizController}: Manages the interactions related to quiz management.
+  \item \textbf{QuizService}: Contains the business logic for quiz management.
+  \item \textbf{QuizRepository}: Responsible for storing and retrieving quiz-related data.
+\end{itemize}
+
+\subsubsection{Question Bank Module}
+
+\begin{itemize}
+  \item \textbf{QuestionController}: Manages interactions related to question management.
+  \item \textbf{QuestionService}: Contains the business logic for question management.
+  \item \textbf{QuestionRepository}: Responsible for storing and retrieving question data.
+\end{itemize}
+
+\subsubsection{Common Processing}
+
+\item \textbf{Data Validation:} Uses React with Formik and Yup for form handling and schema-based validation.
+
+\item \textbf{Error Handling:} Uses ASP.NET Core middleware for global exception handling, logging errors to a centralized logging service (like Serilog) and returning standardized error responses.
+
+\item \textbf{User Authentication:} Manages user authentication and authorization using JWT tokens for securing API endpoints. Middleware checks token validity and user roles before processing quiz-related requests.
+
+\begin{figure}[htbp]
+    \centering
+    \includedrawio[width=\textwidth]{viewpoints/development/development.drawio}
+    \caption{Module Organization Diagram}
+    \label{fig:development}
+\end{figure}
+
+\subsection{Standardization of Design}
+
+\subsubsection{Design Patterns}
+
+\begin{itemize}
+  \item \textbf{Factory Pattern:} Used for creating instances of different quiz types.
+  \item \textbf{Repository Pattern:} Abstracts data access, promoting a clean separation between the business logic and data access layers.
+  \item \textbf{Singleton Pattern:} Ensures a single instance of services like QuizSessionService to manage state across the application.
+\end{itemize}
```

Comment:
Actually be specific. What logging package and how must be used? If you write a new repository, does it have to extend a particular interface (consider simple class diagram showing that important interface). And so on.